pkg/controller/apimanager/upgrade_26_to_27.go

```diff
@@ -16,366 +10,18 @@ type Upgrade26_to_27 struct {
-	if err != nil {
-		return false, err
-	}
-
-	newImageURL := product.CurrentImageProvider().GetZyncPostgreSQLImage()
-	return u.upgradeImageStream(zyncDatabaseImageStream, newImageURL)
-}
-
-func (u *Upgrade26_to_27) highAvailabilityModeEnabled() bool {
-	return u.cr.Spec.HighAvailability != nil && u.cr.Spec.HighAvailability.Enabled
-}
-
-func (u *Upgrade26_to_27) upgradeImageStreams() (reconcile.Result, error) {
-	anImageStreamChanged := false
-	changed, err := u.upgradeAPIcastImageStream()
-	if err != nil {
-		return reconcile.Result{}, err
-	}
-	if changed {
-		u.logger.Info("APIcast ImageStream upgraded")
-		anImageStreamChanged = true
-	}
-
-	changed, err = u.upgradeBackendImageStream()
-	if err != nil {
-		return reconcile.Result{}, err
-	}
-	if changed {
-		u.logger.Info("Backend ImageStream upgraded")
-		anImageStreamChanged = true
-	}
-
-	changed, err = u.upgradeSystemImageStream()
-	if err != nil {
-		return reconcile.Result{}, err
-	}
-	if changed {
-		u.logger.Info("System ImageStream upgraded")
-		anImageStreamChanged = true
-	}
-
-	if !u.highAvailabilityModeEnabled() {
-		changed, err = u.upgradeBackendRedisImageStream()
-		if err != nil {
-			return reconcile.Result{}, err
-		}
-		if changed {
-			u.logger.Info("Backend Redis ImageStream upgraded")
-			anImageStreamChanged = true
-		}
-
-		changed, err = u.upgradeSystemRedisImageStream()
-		if err != nil {
-			return reconcile.Result{}, err
-		}
-		if changed {
-			u.logger.Info("System Redis ImageStream upgraded")
-			anImageStreamChanged = true
-		}
-
-		changed, err = u.upgradeSystemDatabaseImageStream()
-		if err != nil {
-			return reconcile.Result{}, err
-		}
-		if changed {
-			u.logger.Info("System Database ImageStream upgraded")
-			anImageStreamChanged = true
-		}
-	}
-
-	changed, err = u.upgradeSystemMemcachedImageStream()
-	if err != nil {
-		return reconcile.Result{}, err
-	}
-	if changed {
-		u.logger.Info("System Memcached ImageStream upgraded")
-		anImageStreamChanged = true
-	}
-
-	changed, err = u.upgradeZyncImageStream()
-	if err != nil {
-		return reconcile.Result{}, err
-	}
-	if changed {
-		u.logger.Info("Zync ImageStream upgraded")
-		anImageStreamChanged = true
-	}
-
-	changed, err = u.upgradeZyncDatabaseImageStream()
-	if err != nil {
-		return reconcile.Result{}, err
-	}
-	if changed {
-		u.logger.Info("Zync Database ImageStream upgraded")
-		anImageStreamChanged = true
-	}
-
-	return reconcile.Result{Requeue: anImageStreamChanged}, nil
+func (u *Upgrade26_to_27) upgradeAMPImageStreams() (reconcile.Result, error) {
+	// implement upgrade procedure by reconcile procedure
+	baseReconciler := operator.NewBaseReconciler(u.client, u.apiClientReader, u.scheme, u.logger)
+	baseLogicReconciler := operator.NewBaseLogicReconciler(baseReconciler)
+	reconciler := operator.NewAMPImagesReconciler(operator.NewBaseAPIManagerLogicReconciler(baseLogicReconciler, u.cr))
```

Comment:
This is not enough.

There are ImageStreams that are not reconciled with the AMPImagesReconciler:

I think are DBs in general:
 * Memcached
 * System PostgreSQL or System MySQL depending on what has been configured
 * Backend Redis
 * System Redis

Also for System Postgresql/Mysql and System and Backend redis they should be upgraded only if HA is not set

Reply:
Upgrading DB images is much more than upgrading imagestreams.

This first "upgrade" implementation should only be upgrading AMP images

Reply:
I understand